### PR TITLE
Add mixin and template to associations, add PHPStan support for it

### PR DESCRIPTION
**Pull request by Harfusha**
Based on this issues, i added needed associations, and support for phpstan to correctly analyze associations.
https://github.com/phpstan/phpstan/issues/11399
https://github.com/CakeDC/cakephp-phpstan/issues/35

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks good 
Anyone sees a problem with these annotations?

**Comment by othercorey** [comment]
No issues. I can't review the phpstan changes, but mixin seems ok.

File: tests/PHPStan/PhpDoc/TableAssociationTypeNodeResolverExtension.php
Comment (rochamarcelo) [inline tests/PHPStan/PhpDoc/TableAssociationTypeNodeResolverExtension.php:1]: Is this class really needed here? Applications won't be able to use it.

Comment (dereuromark) [reply tests/PHPStan/PhpDoc/TableAssociationTypeNodeResolverExtension.php:1]: Isnt PHPStan using it?
https://github.com/cakephp/cakephp/pull/17827/files#diff-6f19df6a6307a48db0940e6897591fb08776d2db8a6134737aa708defdb5c92dR29

Comment (dereuromark) [reply tests/PHPStan/PhpDoc/TableAssociationTypeNodeResolverExtension.php:1]: Afaik this is only in the tests/ namespace, so for checking this repo, not for user land usage.

Comment (rochamarcelo) [reply tests/PHPStan/PhpDoc/TableAssociationTypeNodeResolverExtension.php:1]: ok, got it.

**Comment by Harfusha** [comment]
How does this look?

**Comment by dereuromark** [comment]
Thank you for your work!
Anything that needs to be merged upstream?

**Comment by Harfusha** [comment]
Cakephp 5 should be fine

**Pull request merged**